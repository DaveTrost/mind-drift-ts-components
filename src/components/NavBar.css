.NavBar {
  margin-top: 2em;
  
  a {
    text-align: center;
    text-decoration: none;
  }

  button {
    border: solid var(--light-blue) 2px;
    border-radius: 8px;
    width: 8em;
    height: 2.3em;
    background-color: white;
    font-family: Montserrat;
    font-weight: 400;
    font-size: 12px;
    margin-left: 1.5em;
    color: var(--dark-blue);
  }
  button:hover {
    background-color: var(--light-blue);
    border: none;
    color: white;
    cursor: pointer;
  }
}

.NavigationFooter {
  display: flex;
  justify-content: center;
  position: fixed;
  z-index: 5;
  bottom: 0;
  background-color: var(--light-grey);
  width: 100vw;
  max-width: 700px;
  height: calc(38px + 2vh);
  a {
    margin: auto;
    margin-left: 11em;
    margin-right: 1em;
  }
  img {
    width: 65%;
    height: auto;
  }
}
