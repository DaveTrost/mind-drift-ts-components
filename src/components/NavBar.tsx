import React from 'react';
import { Link } from 'react-router-dom';
// const { useLogout } = require('../utils/WithSession');
import HOMEicon from '../assets/HOMEicon.png';
import PROFILEicon from '../assets/PROFILEicon.png';
import SETTINGSicon from '../assets/SETTINGSicon.png';
import FLOWERicon from '../assets/FLOWERicon.png';
import './NavBar.scss';


const NavBar = () => {
  // const logout = useLogout();
  // const logout:void = console.log('logout');

  return (
    <div className="NavBar">
      <button>Log Out</button>
      <span className="NavigationFooter">
        <Link to="/"><img src={HOMEicon} alt="Home" />HOME</Link>&nbsp;
        <Link to="/profile"><img src={PROFILEicon} alt="Profile" /></Link>
        <Link to="/settings"><img src={SETTINGSicon} alt="Settings" /></Link>
        <Link to="/about-us"><img src={FLOWERicon} alt="About Us" /></Link>
      </span>
    </div>
  );
};

export default NavBar;
