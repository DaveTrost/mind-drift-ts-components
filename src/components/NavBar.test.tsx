import React from 'react';
import { shallow } from 'enzyme';
import NavBar from './NavBar';

// import useLogoutMock from '../utils/WithSession';
// jest.mock('../utils/WithSession', () => ({}));

describe('Nav Bar component', () => {
  it('renders Nav Bar', () => {
    // useLogoutMock.useLogout = () => {};
    const wrapper = shallow(<NavBar />);
    expect(wrapper).toMatchSnapshot();
  });
});
