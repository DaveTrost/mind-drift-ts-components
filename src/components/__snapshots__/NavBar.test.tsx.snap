// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Bar component renders Nav Bar 1`] = `
<div
  className="NavBar"
>
  <button>
    Log Out
  </button>
  <span
    className="NavigationFooter"
  >
    <Link
      to="/"
    >
      <img
        alt="Home"
        src="test-file-stub"
      />
    </Link>
    Â 
    <Link
      to="/profile"
    >
      <img
        alt="Profile"
        src="test-file-stub"
      />
    </Link>
    <Link
      to="/settings"
    >
      <img
        alt="Settings"
        src="test-file-stub"
      />
    </Link>
    <Link
      to="/about-us"
    >
      <img
        alt="About Us"
        src="test-file-stub"
      />
    </Link>
  </span>
</div>
`;
